.github/workflows/slurm_job_scripts/oci_cluster_manager.py

```diff
@@ -0,0 +1,165 @@
+
+def get_cluster_hostkeys(region):
+    return requests.post(_API_URL, json={'name':'get_cluster_hostkeys', 'region':region},
+                         auth=_API_AUTH, timeout=_REQUEST_TIMEOUT_SECONDS).json()['cluster_hostkeys']
+
+def get_status(region):
+    return requests.post(_API_URL, json={'name':'get_status', 'region': region},
+                         auth=_API_AUTH, timeout=_REQUEST_TIMEOUT_SECONDS).json()['status']
+
+def destroy_all_clusters():
+    logging.debug(requests.post(_API_URL, json={'name':'destroy_all_clusters'},
+                                auth=_API_AUTH, timeout=_REQUEST_TIMEOUT_SECONDS))
+
+# This function is used to create and destroy clusters on demand.
+# A few caveats should be noted:
+# - Depending on resource availability, it might not be possible to create a cluster.
+#   In that case, the script will eventually fail.
+# - Creating a cluster takes time. Similarly, destroying a cluster also takes time.
+#   User should not attempt to concurrently create clusters.
+#   As a rule of thumb, this script should only be used at most once every ~6 hours.
+# - In case a pull-request indirectly calls this script, users should take care to ensure
+#   no other pipeline is attempting to create a cluster at the same time and within a ~3h time window.
+# - Clusters are automatically destroyed 2h after being created, regardless of whether
+#   `destroy_all` is called or not.
+
+def main():
+
+    command_choices = ["create_cluster", "destroy_clusters"]
+    parser = argparse.ArgumentParser(description="Creates and destroy compute clusters on-demand.")
+    parser.add_argument("command", choices=command_choices, \
+        help="""create_cluster will first try to find an existing running cluster,
+                and otherwise attempt to create one in any region.
+                When succesfull, the output will contains the headnode hostkeys, the username and ip address of the cluster, or FAILED.
+                destroy_clusters will destroy all existing clusters.""")
+    parser.add_argument("--pubkey", default=None, \
+                        help='public key to upload to the cluster (relevant only with `create_cluster`).')
+
+    args = parser.parse_args()
+
+    if args.command == "create_cluster":
+
+        found_region = find_existing_cluster()
+        logging.info(f"Found cluster {found_region}")
+
+        if found_region is None:
+
+            logging.info("Could not find existing cluster. Trying to create one.")
+
+            regions = get_regions()
+            logging.info(f"Regions considered: {regions}")
+
+            for region in regions * _RETRY_PER_REGION:
+
+                logging.info(f"Trying region {region}")
+                create_cluster(region)
+                status = get_status(region)
+
+                while status == 'WAIT':
+
+                    logging.info("Waiting...")
+                    time.sleep(_SLEEP_TIME_SECONDS)
+                    status = get_status(region)
```

Comment:
You might wanna implement a timeout here?

Reply:
`get_status(...)`  will always immediately return in normal conditions, so I don't think that's necessary. The http call itself does have a timeout too. So I think we should be OKay.